langchain/src/callbacks/handlers/tracers.ts

```diff
@@ -458,3 +595,123 @@ export class LangChainTracer extends BaseTracer {
     return tracerSession;
   }
 }
+
+export class LangChainTracerV2 extends LangChainTracer {
+  exampleId?: string;
+
+  tenantId?: string;
+
+  constructor(exampleId?: string, tenantId?: string) {
+    super();
+    this.tenantId =
+      tenantId ??
+      (typeof process !== "undefined"
+        ? // eslint-disable-next-line no-process-env
+          process.env?.LANGCHAIN_TENANT_ID
+        : undefined);
+    this.exampleId = exampleId;
+  }
+
+  async updateTenantId(): Promise<string> {
+    const endpoint = `${this.endpoint}/tenants`;
+    const response = await fetch(endpoint, {
+      method: "GET",
+      headers: this.headers,
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to fetch tenant ID: ${response.status} ${response.statusText}`
+      );
+    }
+
+    const tenants = await response.json();
+    if (!tenants || tenants.length === 0) {
+      throw new Error(`No tenants found for endpoint ${endpoint}`);
+    }
+
+    const tenantId = tenants[0].id;
+    this.tenantId = tenantId;
+    return tenantId;
+  }
+
+  protected async _handleSessionResponse(
+    endpoint: string
+  ): Promise<TracerSession | TracerSessionV2> {
+    const tenantId = this.tenantId ?? (await this.updateTenantId());
+    const configured_endpoint = `${endpoint}&tenant_id=${this.tenantId}`;
+    const response = await fetch(configured_endpoint, {
+      method: "GET",
+      headers: this.headers,
+    });
+    let tracerSession: TracerSessionV2;
+    if (!response.ok) {
+      console.error(
+        `Failed to load session: ${response.status} ${response.statusText}`
+      );
+      tracerSession = {
+        id: uuid.v4(),
+        start_time: Date.now(),
+        tenant_id: tenantId,
+      };
+      this.session = tracerSession;
+      return tracerSession;
+    }
+    const resp = (await response.json()) as TracerSessionV2[];
+    if (resp.length === 0) {
+      tracerSession = {
+        id: uuid.v4(),
+        start_time: Date.now(),
+        tenant_id: tenantId,
+      };
+      this.session = tracerSession;
+      return tracerSession;
+    }
+    [tracerSession] = resp;
+    this.session = tracerSession;
+    return tracerSession;
+  }
+
+  private async _convertToCreate(
+    run: RunV2,
+    example_id: string | undefined = undefined
+  ): Promise<RunV2Create> {
+    const persistedRun: RunV2Create = {
+      id: run.id,
+      name: run.name,
+      start_time: run.start_time,
+      end_time: run.end_time,
+      run_type: run.run_type,
+      reference_example_id: example_id,
+      extra: run.extra,
+      execution_order: run.execution_order,
+      serialized: run.serialized,
+      error: run.error,
+      inputs: run.inputs,
+      outputs: run.outputs,
+      session_id: run.session_id,
```

Comment:
There must be a better way of initializing objects like `...run` with everything except 1 or 2 fields. Will look into it

Reply:
Which fields are you trying to remove?

Reply:
Note the backend actually needs to accept (and ignore) extra unknown fields for forwards compatibility (if it doesn't we need to fix that @agola11)

Reply:
Oh OK that makes sense. Really the `parent_run_id` and `child_execution_order` fields

Reply:
extra/unknown fields are already ignored. (This is the default behavior of pydantic btw)

Reply:
Hm If I do 
```ts
const persistedRun: RunV2Create = {
      id: run.id,
      name: run.name,
      start_time: run.start_time,
      end_time: run.end_time,
      run_type: run.run_type,
      extra: run.extra,
      execution_order: run.execution_order,
      serialized: run.serialized,
      error: run.error,
      inputs: run.inputs,
      outputs: run.outputs,
      session_id: run.session_id,
      reference_example_id: example_id,
      child_runs: await Promise.all(
        run.child_runs.map((child_run) => this._convertToCreate(child_run))
      ),
    };
    ```
    it works but if I do
    
    ```
    const persistedRun: RunV2Create = {
      ...run,
      reference_example_id: example_id,
      child_runs: await Promise.all(
        run.child_runs.map((child_run) => this._convertToCreate(child_run))
      ),
    };
```

It returns a 404